### DISCUSSION TITLE
DirectiveType

### DISCUSSION THREAD

**Discussion by bronte2k7** [mirumee/ariadne#807] [Ideas]
I liked the simplicity of ariadne. Using decorators or handler methods makes it easier to understand what's going on. Directives built on classes look very cumbersome.

```python

with_permission = DirectiveType("withPermission")

@with_permission.visit_object
def directive_visit_object(obj, info):
    pass

# or

with_permission.set_visit_object(directive_visit_object)

make_executable_schema(type_defs, resolvers, snake_case_fallback_resolvers, directives=[with_permission])
```

It was a good idea, but for some reason it was not implemented. One style for all code)))

**Comment by rafalp** [comment]
Directives are classes for a reason: they mutate GraphQL schema and changing resolvers is just one of operations that is possible to do with them. This is why we've didn't purse functional API for those.

Example with `directive_visit_object(obj, info)` doesn't make sense because there's no `info` when directives are applied on schema.

**Reply by bronte2k7** [reply]
> Directives are classes for a reason: they mutate GraphQL schema and changing resolvers is just one of operations that is possible to do with them. This is why we've didn't purse functional API for those.
> 
> Example with `directive_visit_object(obj, info)` doesn't make sense because there's no `info` when directives are applied on schema.

It's a pity.

Nevertheless, the implementation in Python looks nicer than in Javascript. thanks for the answer
